### ISSUE TITLE
Release v1.8 on April 16th (anniversary of Ultima VII)

### ISSUE THREAD

**Issue by DominusExult** [exult/exult#176]
I'm preparing a new release in exactly a week on the anniversary of Ultima VII.
Therefore I branched v1.8.x for preparing this. Next step is going to be building these and testing whether they run and the version/copyright dates are really correct on Windows and macOS.

@mbanck, I wonder if you could again test for linux :)

@ceckak agreed to make an Android build, even though his code is not yet completely merged and will only continue to be merged in master - Thanks!

If anyone knows of on an issue that needs to be addressed first or stop this release for any reason, now is the time to speak up! :)

**Comment by mbanck** [comment]
My suggestion would be to tag/release a release candidate in order to test the release procedure and also for packages (e.g. me) to test the tarballs.

**Comment by DominusExult** [comment]
I did a rc1 tag https://github.com/exult/exult/releases/tag/v1.8rc1 - can you test the linux release procedure anyway?

**Comment by ceckak** [comment]
I've spot-checked a rollback of the SDL version for android and confirmed it fixes issues I was seeing with the virtual joystick and music playback.  I'll set up a branch based off this tag in my android fork tonight.

**Comment by i30817** [comment]
BTW, is the gimp plugin working with recent gimp again (ie: not 2.0)?

I have it committed from the my build, which is not ideal.

**Comment by DominusExult** [comment]
Marzo just fixed building the plugin - it was broken for linux/macOS when he converted it to c++. Using gimptool is somehow broken. 
But it works now with current Gimp 2.10.30.

**Comment by Dragon-Baroque** [comment]
> But it works now with current Gimp 2.10.30.

I am afraid I also have some compilation issues on Linux. One of them I reported as a comment to Marzo's commit, and I have two more by now [ trying to compile it on my dev branches ].

Would you agree to a Pull Request ?

**Comment by DominusExult** [comment]
yes, list all that here, I'd say, for discussion (I did NOT understand your comment with placing the header files close to each other) and then you have the power to commit the fixes yourself :)

**Comment by Dragon-Baroque** [comment]
Here are the diffs, with some comments below each one

```
--- a/configure.ac
+++ b/configure.ac
@@ -1060,8 +1060,8 @@ if test x$enable_gimp_plugin = xyes; then
                AC_MSG_RESULT([no, not building GIMP plugin])
        else
                AC_MSG_CHECKING([for GIMP version])
-               gimp_version=`$GIMPTOOL --version | awk 'BEGIN { FS = "."; } { print $1 * 1000 + $2*100+$3;}'`
-               if test "$gimp_version" -ge 2800; then
+               gimp_version=`$GIMPTOOL --version | awk 'BEGIN { FS = "."; } { print $1 * 10000 + $2 * 100 + $3;}'`
+               if test "$gimp_version" -ge 20800; then
                        AC_MSG_RESULT([found >= 2.8.0])
                        AC_SUBST(GIMPTOOL)
                        AM_CONDITIONAL(GIMP_PLUGIN, true)
```
Cosmetic one, but since Gimp is now 2.10, I prefer 210xx than 30xx.
```
--- a/mapedit/u7shp.cc
+++ b/mapedit/u7shp.cc
@@ -31,26 +31,20 @@
 #      pragma GCC diagnostic ignored "-Wdeprecated-declarations"
 #      pragma GCC diagnostic ignored "-Wold-style-cast"
 #      if !defined(__llvm__) && !defined(__clang__)
+#              pragma GCC diagnostic ignored "-Wpedantic"
 #              pragma GCC diagnostic ignored "-Wuseless-cast"
 #      else
+#              pragma GCC diagnostic ignored "-Wc99-extensions"
 #              pragma GCC diagnostic ignored "-Wzero-as-null-pointer-constant"
 #      endif
 #endif    // __GNUC__
-#ifdef USE_STRICT_GTK
-#      define GTK_DISABLE_SINGLE_INCLUDES
-#      define GSEAL_ENABLE
-#      define GNOME_DISABLE_DEPRECATED
-#      define GTK_DISABLE_DEPRECATED
-#      define GDK_DISABLE_DEPRECATED
-#endif    // USE_STRICT_GTK
 #include <gtk/gtk.h>
+#include <libgimp/gimp.h>
+#include <libgimp/gimpui.h>
 #ifdef __GNUC__
 #pragma GCC diagnostic pop
 #endif  // __GNUC__
 
-#include <libgimp/gimp.h>
-#include <libgimp/gimpui.h>
-
 /* Declare some local functions.
  */
 static void   query();
```
Three points in that one :
- Move `gimp.h` and `gimpui.h` near `gtk.h` into the frame that deactivates some warnings by GCC and CLang,
- Add further specific warning deactivates needed by `gimp.h` and `gimpui.h` [ pedantic for GCC, and c99-extensions for CLang ]
- Remove the `USE_STRICT_GTK` section. When the `-DUSE_STRICT_GTK` is not set, like in normal Exult builds, this has no effect, but my builds set it, this was a step in my effort to move to GTK+ 3, but I never worked on upgrading `u7shp.cc` to GTK+ 3, as it was useless until Gimp itself moved to GTK+ 3. The fact is, the Gimp plugin uses a GTK+ 2 widget, the FileSelector, which was _already_ deprecated in GTK+ 2. With `-DUSE_STRICT_GTK` set in my branches, this widget became invisible, and I had compilation errors.

Now would you please tell me how I can commit a fix myself. Please bear with me, this would be my first direct commit. Should I still perform a Pull Request that I approve myself ? Or is my fork of Exult no longer useful and I may act on the reference tree directly with remote git commands under my ID ?

**Comment by DominusExult** [comment]
> Now would you please tell me how I can commit a fix myself. Please bear with me, this would be my first direct commit. Should I still perform a Pull Request that I approve myself ? Or is my fork of Exult no longer useful and I may act on the reference tree directly with remote git commands under my ID ?

Both works :)
But usually you can/should act on the reference tree directly now.
Your fork is still useful if you are unsure about a larger commit and thus can put it up for review in a better way than a diff, IMO.

**Comment by marzojr** [comment]
Both diffs LGTM.

**Comment by Dragon-Baroque** [comment]
Since I had to perform the commit twice [ master + v1.8.x ], I tried with the two ways :
- Against master with a Pull Request that I approved myself,
- Against v1.8.x with a direct commit and push from my clone.

Now I see a few commits authored by X but commited by Y. How do you get this ? Cherry pick perhaps ?

**Comment by DominusExult** [comment]
Yes, I did that by „git cherry-pick hash“. This keeps the name of the original committer and a reference where it comes from.
In the case of the v1.8.x branch it’s better than rebasing/merging as I backed out the "exult-as-library" commit in this branch.

**Comment by DominusExult** [comment]
Congrats on your first direct commit ;)

**Comment by Dragon-Baroque** [comment]
Question : How do I add branch v1.8.x to my fork to track v1.8.x of reference ?

**Comment by marzojr** [comment]
Assuming you have two remotes configured (say, `origin` for your fork and `upstream` for the main Exult repo), you do something like:

```
git fetch upstream
git checkout v1.8.x
git push origin v1.8.x
```

**Comment by Dragon-Baroque** [comment]
Worked perfectly.
Thank you.

**Comment by ceckak** [comment]
Created an [android-port-v1.8.x branch in my fork](https://github.com/ceckak/exult/tree/android-port-v1.8.x) which I will keep rebased against the v1.8.x official branch.  Once you're ready to lock down a final tag, I'll make a release build of the apk.

**Comment by Dragon-Baroque** [comment]
@DominusExult, you suggested one month ago that I should wrap up my Audio change #88 into a Pull Request for integration into the 1.8.0. I could not last month because of my involvement in a Choir concert that took place end of March, and out of which I am emerging only now. Was a good success, fortunately...

 I suppose that making the Pull Request _today_ would be much too late for 1.8.0, yes ?

Moreover, I am less convinced than one year ago about the worth of that Audio change.

I would then postpone #88 to 1.8.1, along with :

- An upgrade of the Gimp plugin to use _non deprecated_ GTK+ 2. So that we could in confidence reactivate the `USE_STRICT_GTK`.
-  Some further thought about the Audio in Exult. Such as going fully digital _by default_ on workstations ( MacOS, Win32, Linux, but not iOS or Android ).

**Comment by DominusExult** [comment]
Yes, it's too late now for 1.8. As soon as the release gets done, I'll make master into 1.9.0 and that can be used for "toying" (you can even toy with that now :)).
1.8.x will only be used for urgent bug fixes but we might not even go there. We didn't for 1.6 so we probably will direct people at the 1.9 snapshots, again

**Comment by Dragon-Baroque** [comment]
Agreed.

Then going back to master, I would like either to revert #174 commit  c1779bf899a5d6359f768437409b47f22fc752be, like you did on 1.8.x, or complete it with something like 
```
 libexult_la_LIBADD = \
-       $(EXULTLIBS) $(PNG_LIBS) \
+       $(EXULTLIBS) $(PNG_LIBS) $(FREETYPE2_LIBS) \
        $(SDL_LIBS) $(SYSLIBS) $(x_libraries) $(ICON_FILE) \
        $(ZLIB_LIBS) $(VORBIS_LIBS) $(VORBISFILE_LIBS) $(OGG_LIBS) $(MT32EMU_LIBS) $(FLUIDSYNTH_LIBS) $(ALSA_LIBS)
 
 libexult_la_DEPENDENCIES = $(ICON_FILE) $(EXULTLIBS)
 
+libexult_la_LDFLAGS = -release 1.8.0
+
```
for at this moment, building master with `--enable-shared` fails.

**Comment by DominusExult** [comment]
I think for now, you should just revert c1779bf. I've been banging my head against the wall on how to do this another way but since all that in makefile.am is happening *before* configure sets the OS I've not been successful.

**Comment by ceckak** [comment]
Go ahead and revert it for now.  I've been experimenting with a way to have a configure flag to specify whether you want to build exult as an executable or a library.  This is actually what I wanted to do originally, but it's harder to make it work than I'd like, and that's how I ended up with what you currently have.  I'll come up with something though and make a new pull request.

I would eventually be interested in exploring whether the library could be helpful for the ios build.  I could have just set up an android project and added all the exult sources to it, but I went out of my way to avoid that so that I wouldn't have to maintain the android project in parallel with the core exult build environment.  The way I have it set up currently, I can use the normal autotools build to configure and build exult as a library, and then the android project itself just references the android-specific sources and pulls in the common exult runtime library.  I don't have any experience with ios builds, but we might be able to do something similar there if it sounds useful.

**Comment by DominusExult** [comment]
The iOS build is just using an xcode project file and not the autotools at all. With the "sub" project files for the libs it's just one button to hit to build Exult, to test in the iOS simulator, quickly get it on an iOS device or submit it to Apple's TestFlight program. The initial port to iOS used the autotools but it was a hassle to get it onto a device.
in theory your way sounds interesting, though, I do understand how it might be beneficial.

**Comment by ceckak** [comment]
Understand, and ultimately, whatever is preferred/easiest for the ios maintainer is the best solution.  I tend to be allergic to IDEs and maintaining parallel build configurations.

In any case, I think I have a solution working locally now which introduces a new configure flag that builds the library instead of the executable.  It would be disabled by default, but enabled during the android build, and available for use by any other builds that want it in the future.

I still need to do a little more work to clean it up and incorporate the library version as @Dragon-Baroque suggested.  Once I have it cleaned up, I'll post a new pull request and you guys can look it over.

Back on topic, whenever you get the final 1.8 tag applied, I'll start working on the android build.

**Comment by DominusExult** [comment]
Tagged v1.8 just now.

Reverted the exult-as-lib on master, too. 

I think most of us are allergic to IDEs, I'm doing all macOS stuff with autotools. Just for iOS, with the goal to have it at least on Apple's TestFlight, I found no easy way around Xcode. Maybe if Apple is forced to allow sideloading...

**Comment by drcode1** [comment]
Cheers!  :-)
